plugins/cortex/pkg/cortex/admin.go

```diff
@@ -355,6 +356,23 @@ func (p *Plugin) GetRule(ctx context.Context,
 	}, nil
 }
 
+func (p *Plugin) ListRules(ctx context.Context, req *cortexadmin.Cluster) (*cortexadmin.QueryResponse, error) {
+	lg := p.logger.With(
+		"cluster id", req.ClusterId,
+	)
+	resp, err := listCortexRules(p, lg, ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
```

Comment:
Should add additional info to these errors, otherwise you might just get like "error: EOF"

Reply:
Done